l5kit/l5kit/cle/composite_metrics.py

```diff
@@ -130,3 +132,81 @@ def compute(self, metric_results: Dict[str, torch.Tensor],
         driven_miles = metric_results[self.driven_miles_metric_name].sum()
         driven_miles_cpu = driven_miles.cpu().item()
         return float(driven_miles_cpu)
+
+
+class SupportsCompositeMetricAggregation(Protocol):
+    """Protocol supporting the composite metric aggregation. This aggregator
+    is responsible for aggregating results from the the composite metrics and
+    also doing a reduction across multiple distributed nodes.
+    """
+
+    def aggregate(self, scene_composite_metric_results:
+                  Dict[int, Dict[str, float]]) -> Dict[str, Any]:
+        """This method will aggregate scenes locally and then will
+        do the reduction step to aggregate data across distributed
+        nodes.
+
+        :param scene_composite_metric_results: results from composite metric
+                                               outputs per scene
+        :returns: any result indexed by the composite metric name.
+        """
+        raise NotImplementedError
+
+
+class ReduceMode(IntEnum):
+    """Reduction modes supported by the composite metric aggregator."""
+    #: Summation of results locally and across nodes
+    SUM = 0
+
+
+class CompositeMetricAggregator(SupportsCompositeMetricAggregation):
+    """Aggregates composite metric results locally and across multiple
+    distributed nodes.
+
+    :param reduce_mode: the reduce operation to be applied both into
+                        local scenes and also across distributed
+                        nodes.
+    """
+
+    #: Mapping from ReduceMode to torch distributed ops
+    torch_reduce_mapping = {
+        ReduceMode.SUM: torch.distributed.ReduceOp.SUM
+    }
+
+    def __init__(self, reduce_mode: ReduceMode = ReduceMode.SUM):
+        # Supporting only ReduceMode.SUM for the moment
+        if reduce_mode != ReduceMode.SUM:
```

Comment:
shouldn't we check in `ReduceMode` ?

Reply:
Not sure if I understood, check where you mean ?

Reply:
Shouldn’t we check if `reduce_mode` is among those in `ReduceMode`?

Reply:
Ah ok, makes sense, will fix it.